### PR TITLE
feat: add get session replay link API

### PR DESCRIPTION
**Pull request by ahmedAlaaInstabug**
## Description of the change

1. Map OnSessionReplayLinkReady API that Retrieves the current session's replay link.

## Type of change

- [ ] Bug fix (non-breaking change that fixes an issue)
- [x] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Related issues

Jira ID = [MOB-14010](https://instabug.atlassian.net/browse/MOB-14010)

### Development

- [ ] Lint rules pass locally
- [ ] The code changed/added as part of this pull request has been covered with tests

### Code review

- [ ] This pull request has a descriptive title and information useful to a reviewer
- [ ] Issue from task tracker has a link to this pull request

[MOB-14010]: https://instabug.atlassian.net/browse/MOB-14010?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by mnoseir** [comment]
### Coverage Report
| Label | Coverage | Status |
| ----- | -------- | ------ |
| JavaScript | 96.3% | ✅ |
| Android | 47.5% | ✅ |
| iOS | 41.5% | ✅ |
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against c535cead22e2f4d15496d105c63b2939b3493992
</p>

File: test/modules/SessionReplay.spec.ts [hunk 1]
```diff
@@ -29,4 +29,10 @@ describe('Session Replay Module', () => {
     expect(NativeSessionReplay.setUserStepsEnabled).toBeCalledTimes(1);
     expect(NativeSessionReplay.setUserStepsEnabled).toBeCalledWith(true);
   });
+
+  it('should call the native method getSessionReplayLink', () => {
+    SessionReplay.getSessionReplayLink();
+
+    expect(NativeSessionReplay.getSessionReplayLink).toBeCalledTimes(1);
```
Comment (a7medev) [inline test/modules/SessionReplay.spec.ts:36]: I think we also need to assert that it returns whatever value `NativeSessionReplay.getSessionReplayLink` returns which means we also need to mock the returned value of `NativeSessionReplay.getSessionReplayLink` then verify that `SessionReplay.getSessionReplayLink` returns the same value.

What do you think?

**Review by a7medev — CHANGES_REQUESTED** [review]
Great job, Alaa 👏🏼! Can you please check the comments I've added? 🙏🏼

**Review by abdelhamid-f-nasser — APPROVED** [review]
@ahmedAlaaInstabug Great effort and yet another great feature 👏 💪 !!!

**Pull request merged**